pallets/manta-pay/src/lib.rs

```diff
@@ -486,21 +486,18 @@ pub mod pallet {
 		#[inline]
 		fn pull_receivers_for_shard(
 			shard_index: u8,
-			receiver_index: &mut usize,
+			receiver_index: usize,
 			receivers: &mut ReceiverChunk,
 		) -> bool {
-			let mut iter = if *receiver_index == 0 {
+			let mut iter = if receiver_index == 0 {
 				Shards::<T>::iter_prefix(shard_index)
 			} else {
-				let raw_key = Shards::<T>::hashed_key_for(shard_index, *receiver_index as u64 - 1);
+				let raw_key = Shards::<T>::hashed_key_for(shard_index, receiver_index as u64 - 1);
 				Shards::<T>::iter_prefix_from(shard_index, raw_key)
 			};
 			for _ in 0..Self::PULL_MAX_PER_SHARD_UPDATE_SIZE {
 				match iter.next() {
-					Some((_, (utxo, encrypted_note))) => {
-						*receiver_index += 1;
-						receivers.push((utxo, encrypted_note));
-					}
+					Some((_, next)) => receivers.push(next),
```

Comment:
Is it possible that hackers use it to attack us? I mean the loop and allocation happen to runtime, if there're too many requests sent, like 10K or even more, runtime might be exploded. Of cause we could set limits on rpc request like only 100/sec, and only enable on full node.

Reply:
The is outside of the consensus. For example, collator should have RPC disabled.

Reply:
We do need to do a proper DDoS tests tho.

Reply:
> We do need to do a proper DDoS tests tho.

Definitely.

Reply:
https://github.com/Manta-Network/Manta/issues/587